ai2thor/controller.py

```diff
@@ -519,7 +521,22 @@ def key_sort_func(scene_name):
 
         self.server.send(dict(action='Reset', sceneName=scene, sequenceId=0))
         self.last_event = self.server.receive()
-        
+
+        # update the initialization parameters
+        init_params = init_params.copy()
+
+        # width and height are updates in 'ChangeResolution', not 'Initialize'
+        if 'width' in init_params or 'height' in init_params:
+            if 'width' in init_params:
+                self.width = init_params['width']
+                del init_params['width']
+            if 'height' in init_params:
+                self.height = init_params['height']
+                del init_params['height']
+            self.step(action='ChangeResolution', x=self.width, y=self.height)
```

Comment:
Can we not run `ChangeResolution` if the resolution is already correct? I've had weird issues with `ChangeResolution` in the past and so want to run it as few times as possible.

Reply:
Agree. It already shouldn't run if it's already correct. It's inside of the `if 'width' in init_params or 'height' in init_params:` condition.

Reply:
Actually can also add if `width == self.width` and `height == self.height` do not run.

Reply:
Updated with https://github.com/allenai/ai2thor/pull/559/commits/c2cfbafbfc36a927d0d9e592be30951c06c35617